services/static-webserver/client/source/class/osparc/auth/ui/RequestAccount.js

```diff
@@ -240,41 +245,102 @@ qx.Class.define("osparc.auth.ui.RequestAccount", {
-      osparc.FlashMessenger.getInstance().logAs(msg, "INFO");
-      this.fireDataEvent("done");
+    __requestPressed: function() {
+      const validForm = this._form.validate();
+      if (validForm) {
+        const formData = {};
+        Object.entries(this._form.getItems()).forEach(([key, field]) => {
+          const val = field.getValue();
+          if (val && (typeof val === "object") && ("classname" in val)) {
+            formData[key] = val.getModel();
+          } else {
+            formData[key] = val;
+          }
+        });
+        this.__submit(formData, this.__captchaField.getValue());
+      }
+    },
 
+    __submit: function(formData, captchaValue) {
       const params = {
         data: {
-          "form": formData
+          "form": formData,
+          "captcha": captchaValue
         }
       };
-      osparc.data.Resources.fetch("auth", "postRequestAccount", params);
+      osparc.data.Resources.fetch("auth", "postRequestAccount", params)
+        .then(() => {
+          const msg = this.tr("The request is being processed, you will hear from us in the coming hours");
+          osparc.FlashMessenger.getInstance().logAs(msg, "INFO");
+          this.fireDataEvent("done");
+        })
+        .catch(err => {
+          console.error(err);
+          osparc.FlashMessenger.logAs(err.message, "ERROR");
+          this.__restartCaptcha();
+        });
+    },
+
+    __createCaptchaLayout: function() {
+      const captchaGrid = new qx.ui.layout.Grid(5, 5);
+      captchaGrid.setColumnAlign(0, "center", "bottom");
+      captchaGrid.setColumnFlex(1, 1);
+      const captchaLayout = new qx.ui.container.Composite(captchaGrid);
+
+      const captchaImage = this.__captchaImage = new qx.ui.basic.Image().set({
+        allowShrinkX: true,
+        allowShrinkY: true,
+        scale: true,
+        width: 140,
+        height: 45,
+        cursor: "pointer"
+      });
+      this.__captchaImage.addListener("tap", () => this.__restartCaptcha(), this);
+      captchaLayout.add(captchaImage, {
+        column: 0,
+        row: 0,
+        rowSpan: 2
+      });
+
+      const label = new qx.ui.basic.Label(this.tr("Type the 6 digits:")).set({
+        font: "text-12"
+      });
+      captchaLayout.add(label, {
+        column: 1,
+        row: 0
+      });
+
+      const captchaField = this.__captchaField = new qx.ui.form.TextField().set({
+        backgroundColor: "transparent",
+        required: true
+      });
+      captchaLayout.add(captchaField, {
+        column: 1,
+        row: 1
+      });
+
+      return captchaLayout;
+    },
+
+    __restartCaptcha: function() {
+      this.__captchaImage.setSource(null);
+      let url = osparc.data.Resources.resources["auth"].endpoints["captcha"].url;
+      // Since the url doesn't change, this dummy query parameter will force the frontend to make a new request
+      url += "?" + Math.floor(Math.random() * 100);
```

Comment:
OK, so you are sending a fake query just to avoid the front-end to cache the response? looks like a bit dirty trick :-) Isn;t there a way to reset or disable cache?

Reply:
When doing this, you usually use a datetime (more reliable) than a random number. I would go for that

Reply:
Yep, DateTime added, thanks 👍